threedi_modelchecker/checks/other.py

```diff
@@ -1082,3 +1068,77 @@ def _get_records(self, session):
                 )
             )
         )
+
+
+class UsedSettingsPresentCheck(BaseCheck):
+    def __init__(
+        self,
+        column,
+        settings_table,
+        filters=None,
+        level=CheckLevel.ERROR,
+        error_code=0,
+    ):
+        super().__init__(column, filters, level, error_code)
+        self.settings_table = settings_table
+
+    def get_invalid(self, session: Session) -> List[NamedTuple]:
+        # more than 1 row should be caught by another check
+        all_results = self.to_check(session).filter(self.column == True).all()
+        use_cols = len(all_results) > 0
+        if use_cols and session.query(self.settings_table).count() == 0:
+            return all_results
+        return []
+
+    def description(self) -> str:
+        return f"{self.column_name} in {self.table.name} is set to True but {self.settings_table.__tablename__} is empty"
+
+
+class SettingsLengthCheck(BaseCheck):
+    def __init__(
+        self,
+        column,
+        filters=None,
+        level=CheckLevel.ERROR,
+        error_code=0,
+        min_length=0,
```

Comment:
Absolutely on board with making a generic, testable check here. However, you added a feature that you don't use (`min_length`/`max_length`). Unless I am missing something? That is basically a waste of effort. I think you should remove the feature.

Maybe call it `MaxOneRecordCheck`?